### PR TITLE
CORE-456: Implement crypto primitive in Java Core Crypto

### PR DESCRIPTION
**Pull request by michael-brd**

### CODE REVIEW

File: Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/Encrypter.java [hunk 1]
```diff
@@ -7,35 +7,57 @@
...
+
 /* package */
 final class Encrypter implements com.breadwallet.crypto.Encrypter {
 
-    // TODO(fix): Implement me!
-
     /* package */
     static Encrypter createForAesEcb(byte[] key) {
-        return null;
+        BRCryptoCipher cipher = BRCryptoCipher.createAesEcb(key).orNull();
```
Comment (EBGToo) [inline Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/Encrypter.java]: I think we'd plan on renaming this class.  Worth doing now?

File: Java/CoreNative/src/main/java/com/breadwallet/corenative/crypto/BRCryptoHasher.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkState;
+
+public class BRCryptoHasher extends PointerType {
+
+    private static final int CRYPTO_HASHER_SHA1         = 0;
```
Comment (EBGToo) [inline Java/CoreNative/src/main/java/com/breadwallet/corenative/crypto/BRCryptoHasher.java]: These need to agree with the C code?  If so, perhaps a comment; internal enum?

File: Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/Encrypter.java (hunk 1 above)
Comment (michael-brd) [reply Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/Encrypter.java]: Certainly. Do you want the Swift updated as well? Will cause a (easily resolvable) break in the iOS app.

**Comment by michael-brd** [comment]
@EBGToo, I've updated to address the comments. Let me know if you are good with me merging in (small iOS breakage due to Encrypter->Cipher change).

**Comment by EBGToo** [comment]
> @EBGToo, I've updated to address the comments. Let me know if you are good with me merging in (small iOS breakage due to Encrypter->Cipher change).

Yes, good.

**Pull request merged**